ddtrace/internal/context_manager.py

```diff
@@ -16,6 +16,10 @@
 
 
 class BaseContextManager(six.with_metaclass(abc.ABCMeta)):
+    @abc.abstractmethod
```

Comment:
I would have this not be abstract and instead define a base `def reset(self):` here if we can

e.g.

```python
def __init__(self, reset=False):
    if reset:
        self.reset()


def reset(self):
    pass
```


Then we'd need to make sure to call `super(Cls, self).__init__(reset=True)` in base classes, and override `def reset(self):` on those that need it (maybe they should all define it anyways?)

Reply:
👍